src/components/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import {classNames} from '@shopify/react-utilities/styles';
+import {createUniqueIDFactory} from '@shopify/javascript-utilities/other';
+import Labelled from '../Labelled';
+import Icon from '../Icon';
+
+import * as styles from './Toggle.scss';
+
+export interface Props {
+  /** Toggles between enabled and disabled */
+  checked?: boolean;
+  /** Disables the toggle, disallowing interaction */
+  disabled?: boolean;
+  /** A unique identifier for the toggle */
+  id?: string;
+  /** Label for the toggle */
+  label: string;
+  /** Visually hide the label */
+  labelHidden?: boolean;
+  /** Element to display before the input */
+  prefix?: React.ReactNode;
+  /** Element to display after the input */
+  suffix?: React.ReactNode;
+  /** Callback when the toggle value is changed */
+  onChange(newChecked: boolean): void;
+}
+
+const getUniqueID = createUniqueIDFactory('Toggle');
+
+export default function Toggle({
+  id = getUniqueID(),
+  checked,
+  disabled,
+  onChange,
+  label,
+  labelHidden,
+  prefix,
+  suffix,
+}: Props) {
+  const className = classNames(
+    styles.Toggle,
+    checked && styles.ToggleChecked,
+    disabled && styles.ToggleDisabled,
+  );
+
+  const inputClassName = classNames(
+    styles.Input,
+    checked && styles.ToggleChecked,
+    disabled && styles.ToggleDisabled,
+  );
+
+  function handleChange(event: React.MouseEvent<HTMLElement>) {
+    if (disabled) {
+      return;
+    }
+
+    const buttonNotChecked =
+      event.currentTarget.getAttribute('aria-checked') === 'true';
```

Comment:
If I understand this correctly it is to call the callback with what the new value will be. We shouldn’t have an aria attribute as the source of truth for this and would expect it to be the checked prop. Something like:

```js
function handleChange(event: React.MouseEvent<HTMLElement>) {
  if (disabled) {
    return;
  }
  
  const newValue = !checked;
  onChange(newValue);
}